src/Components/TodoItem.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onTodoDelete?: () => void;
+  onRenameTodo?: (todoTitle: string) => void;
+  onTodoToggle?: () => Promise<void>;
+  isProcessing: boolean;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onTodoDelete = () => { },
+  onRenameTodo = () => { },
+  onTodoToggle = () => { },
+  isProcessing,
+}) => {
+  const { title, completed, id } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [todoTitle, setTodoTitle] = useState(todo.title);
+
+  const handleTodoDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleTodoSave = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    try {
+      const trimmedTitle = todoTitle.trim();
+
+      if (!trimmedTitle) {
+        setIsEditing(false);
+        await onTodoDelete();
+
+        return;
+      }
+
+      if (todo.title === trimmedTitle) {
+        setIsEditing(false);
+
+        return;
+      }
+
+      if (trimmedTitle) {
+        await onRenameTodo(todoTitle);
+        setIsEditing(false);
+      } else {
+        await onTodoDelete();
+      }
+
+      setIsEditing(false);
+    } catch {
+      throw new Error();
```

Comment:
You don't need to throw new Error here, because you don't have any other `catch`. Please check other places in the code for similar mistakes.